### ISSUE TITLE
MDSplus String data Python3 error

### ISSUE THREAD

**Issue by VadimNV** [MDSplus/mdsplus#1908]
Dear MDSplus team,

  We've spotted a bug in the way the latest MDSplus reports back the string type to Python3.

EXAMPLE:
from MDSplus import Connection
import numpy as np
import sys

import MDSplus
print("MDSplus version "+str(MDSplus.__version__))
print("numpy   version "+str(np.__version__))
p_v = str(sys.version).split('.')[0]
print("python version "+str(p_v))

MDSplus_IP_address = '192.168.1.7:8000'

pulseNo = 7003
conn = Connection(MDSplus_IP_address)
conn.openTree('st40',pulseNo)

place_to_read = 'mag.ROG.ALL:NAME'
data = conn.get(place_to_read).getData().value_of().data() 

test = ('ROG_INIVC000_SPR' in data[1])
print("THIS FAILS DUE TO A BUG: ",test)

OUTPUT DUE TO A BUG

MDSplus version 7.49.3
numpy   version 1.16.2
python version 3
Traceback (most recent call last):
  File "mdsplus_quick.py", line 29, in <module>
    test = ('ROG_INIVC000_SPR' in data[1])
TypeError: a bytes-like object is required, not 'str'

BUG MEDIATION VIA TYPE-CASTING:
if int(p_v)==3:
    data = data.astype(np.str_) # To use actual strings in Python 3 use np.str_ or np.unicode_
    # https://docs.scipy.org/doc/numpy/reference/arrays.dtypes.html

EXPLANATION from https://docs.scipy.org/doc/numpy/reference/arrays.dtypes.html
In Python 2 data type for strings were the equivalent string_ and bytes_. 

In Python 3 data type for strings is the equivalent str_ and unicode_. 

For backward compatibility, if Python 3 **receives an input data marked as type string_ (or "string" or similar - not sure), it maps it to bytes_. However, bytes_ in Python 3 does not support certain string operations, e.g.
a     = np.asarray(['lala','tata']).astype(np.string_)
test = ("lala" in a[0]) # will give us an error

Another important thing is that the above bug is **not present** in combination MDSplus 7.7.9 + Python 3. It is only observed when using a newer version, MDSplus 7.49.3 + Python 3

It seems like the correct behaviour for the MDSplus python interface is to return the string type as str_ (or unicode_) for Python 3 and string_ (or byte_) for Python 2.

Thanks in advance for looking into this. I am looking forward to hearing your thoughts.

Regards,
Vadim

**Comment by zack-vii** [comment]
thanks for filing a report. the example however is of limited usesince it requires access to data that we csnot see. i assume the node contains an array of strings? i will eventually look into this by monday. but as a general advice for future reports. please provide general examples ideally with a log of execution. this way, it is much easier to reproduce the issue.

**Comment by zack-vii** [comment]
sorry but i cannot reproduce this
alpha 7.96.5 with python 3.5.3
```python3
In [1]: import MDSplus
In [2]: a=MDSplus.Array(['asd','qwe'])
In [3]: a[0]
Out[3]: 'asd'
In [4]: a[1]
Out[4]: 'qwe'
In [5]: type(a)
Out[5]: MDSplus.mdsarray.StringArray
In [6]: 'a' in a[0]
Out[6]: True
In [7]: 'asd' in a
Out[7]: True
```

**Comment by zack-vii** [comment]
seemes, it has been resolved since 7.49.3
